x/mongo/driver/topology/pool.go

```diff
@@ -854,7 +882,53 @@ func (p *pool) clear(err error, serviceID *primitive.ObjectID) {
 		}
 		p.lastClearErr = err
 		p.stateMu.Unlock()
+	}
+
+	if mustLogPoolMessage(p) {
+		keysAndValues := logger.KeyValues{
+			logger.KeyServiceID, serviceID,
+		}
+
+		logPoolMessage(p, logger.ConnectionPoolCleared, keysAndValues...)
+	}
+
+	if sendEvent && p.monitor != nil {
+		event := &event.PoolEvent{
+			Type:      event.PoolCleared,
+			Address:   p.address.String(),
+			ServiceID: serviceID,
+			Error:     err,
+		}
+		if interruptAllConnections {
+			event.Interruption = true
+		}
```

Comment:
Optional: No need for a conditional block here.

```suggestion
		event := &event.PoolEvent{
			Type:         event.PoolCleared,
			Address:      p.address.String(),
			ServiceID:    serviceID,
			Error:        err,
			Interruption: interruptAllConnections
		}
```